### PR TITLE
Implement changes to current activity loging [SCI-3021]

### PR DESCRIPTION
**Pull request by mlorb**
Jira ticket: [SCI-3021](https://biosistemika.atlassian.net/browse/SCI-3021)

### What was done
_Existing activities were refactored: [list](https://docs.google.com/spreadsheets/d/182zm7Q1it1U2ggQ9T28r3SKTnoFQFKiiE4hc9Io94Zc/edit#gid=989483893)._

### Note:
_`revert `method in protocols controller was not working. The bug was fixed._

_Also it was decited that all activities in demo project should be removed._

### CODE REVIEW

File: app/controllers/steps_controller.rb [hunk 1]
```diff
@@ -276,41 +252,34 @@ def checklistitem_state
...
-              experiment: @protocol.my_module.experiment,
-              my_module: @protocol.my_module,
-              message: message,
-              type_of: if checked
-                         :check_step_checklist_item
-                       else
-                         :uncheck_step_checklist_item
-                       end
-            )
+            Activities::CreateActivityService
```
Comment (urbanrotnik) [inline app/controllers/steps_controller.rb]: ditto

File: app/controllers/steps_controller.rb [hunk 2]
```diff
@@ -340,30 +309,25 @@ def toggle_step_state
...
           if @protocol.in_module?
-            Activity.create(
-              user: current_user,
-              project: @protocol.my_module.experiment.project,
-              experiment: @protocol.my_module.experiment,
-              my_module: @protocol.my_module,
-              message: message,
-              type_of: completed ? :complete_step : :uncomplete_step
-            )
+            Activities::CreateActivityService
```
Comment (urbanrotnik) [inline app/controllers/steps_controller.rb]: ditto

File: app/helpers/notifications_helper.rb [hunk 3]
```diff
@@ -32,9 +32,10 @@ def generate_notification(user, target_user, team, role, project)
       message = "#{I18n.t('search.index.team')} #{team.name}"
     elsif project
       title = I18n.t('activities.unassign_user_from_project',
-                     unassigned_user: target_user.full_name,
+                     user_target: target_user.full_name,
```
Comment (urbanrotnik) [inline app/helpers/notifications_helper.rb]: 👍

File: spec/controllers/experiments_controller_spec.rb [hunk 4]
```diff
@@ -67,4 +83,18 @@
       end
     end
   end
+
+  describe '#archive' do
+    let(:params) do
+      { id: experiment.id,
+        experiment: { name: 'test experiment A1',
```
Comment (urbanrotnik) [inline spec/controllers/experiments_controller_spec.rb]: Need those params for `archive` action?

File: spec/controllers/project_comments_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ProjectCommentsController, type: :controller do
+  login_user
+
+  let(:user) { User.first }
```
Comment (urbanrotnik) [inline spec/controllers/project_comments_controller_spec.rb]: `User.first` actually works here bc you have `current_user` but it is not intuitive what `User.first` is. So change it to `subject.current_user`.

File: spec/controllers/reports_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ReportsController, type: :controller do
+  login_user
+
+  let(:user) { User.first }
```
Comment (urbanrotnik) [inline spec/controllers/reports_controller_spec.rb]: ditto

File: spec/controllers/result_assets_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ResultAssetsController, type: :controller do
+  login_user
+
+  let(:user) { User.first }
```
Comment (urbanrotnik) [inline spec/controllers/result_assets_controller_spec.rb]: ditto

File: spec/controllers/result_comments_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ResultCommentsController, type: :controller do
+  login_user
+
+  let(:user) { User.first }
```
Comment (urbanrotnik) [inline spec/controllers/result_comments_controller_spec.rb]: ditto

File: spec/controllers/result_comments_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  let!(:team) { create :team, :with_members }
```
Comment (urbanrotnik) [inline spec/controllers/result_comments_controller_spec.rb]: you are creating team here but `user` is not a member of it  I think. Just FYI

File: spec/controllers/result_tables_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ResultTablesController, type: :controller do
+  login_user
+
+  let(:user) { User.first }
```
Comment (urbanrotnik) [inline spec/controllers/result_tables_controller_spec.rb]: ditto

File: spec/controllers/result_tables_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+  let!(:team) { create :team, :with_members }
```
Comment (urbanrotnik) [inline spec/controllers/result_tables_controller_spec.rb]: ditto

File: spec/factories/result_comment.rb [hunk 12]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
```
Comment (urbanrotnik) [inline spec/factories/result_comment.rb]: Factroy should be in plural. So `result_comments.rb` should be better and easier to note the difference to model.

File: spec/factories/result_table.rb [hunk 13]
```diff
@@ -3,5 +3,6 @@
 FactoryBot.define do
   factory :result_table do
     table { create(:table) }
+    result { create(:result) }
```
Comment (urbanrotnik) [inline spec/factories/result_table.rb]: of `result` is association with model named `result` you don't need `{ ... }`

File: app/controllers/my_modules_controller.rb [hunk 14]
```diff
@@ -337,21 +304,6 @@ def assign_samples
         my_module.samples.push(*new_samples)
         task_names << my_module.name
       end
-      if new_samples.any?
-        Activity.create(
-          type_of: :assign_sample,
```
Comment (aignatov-bio) [inline app/controllers/my_modules_controller.rb]: Do we need log this activity?

File: app/controllers/my_modules_controller.rb [hunk 15]
```diff
@@ -375,21 +327,6 @@ def unassign_samples
         task_names << my_module.name
         my_module.samples.destroy(samples & my_module.samples)
       end
-      if samples.any?
-        Activity.create(
-          type_of: :unassign_sample,
```
Comment (aignatov-bio) [inline app/controllers/my_modules_controller.rb]: Do we need log this activity?

**Comment by urbanrotnik** [comment]
@mlorb try to solve the conflict

File: app/controllers/canvas_controller.rb [hunk 16]
```diff
@@ -183,14 +183,13 @@ def update
...
       next if my_module.blank?
-      Activity.create(type_of: :archive_module,
-        project: my_module.experiment.project,
-        experiment: my_module.experiment,
-        my_module: my_module,
-        user: current_user,
-        message: t('activities.archive_module',
-                   user: current_user.full_name,
-                   module: my_module.name))
+      Activities::CreateActivityService
```
[the remark this answers is not in the document]
Comment (Ducz0r) [reply app/controllers/canvas_controller.rb]: I would keep it here, for now. I agree that it's possibly not the best place, but I would stray from refactoring too much in scope of this task.

File: app/controllers/protocols_controller.rb [hunk 17]
```diff
@@ -1234,4 +1211,17 @@ def log_activity(type_of)
               protocol: @protocol.id
             })
   end
+
+  def log_activity_task_protocol(type_of)
```
[the remark this answers is not in the document]
Comment (Ducz0r) [reply app/controllers/protocols_controller.rb]: Yeah, leave it as it is for now.

**Comment by Ducz0r** [comment]
Regarding samples, we can ignore/remove any logic that touches them, at least in scope of the activities. That is legacy code not used any more.

File: app/controllers/canvas_controller.rb (hunk 16 above)
[the remark this answers is not in the document]
Comment (mlorb) [reply app/controllers/canvas_controller.rb]: :+1:

File: app/controllers/my_modules_controller.rb [hunk 18]
```diff
@@ -760,4 +662,26 @@ def my_module_params
     params.require(:my_module).permit(:name, :description, :due_date,
                                       :archived)
   end
+
```
[the remark this answers is not in the document]
Comment (mlorb) [reply app/controllers/my_modules_controller.rb]: :+1:

File: app/controllers/projects_controller.rb [hunk 19]
```diff
@@ -164,43 +157,32 @@ def update
...
-          message: message_visibility
-        )
+        Activities::CreateActivityService
+          .call(activity_type: :change_project_visibility,
+                owner: current_user,
+                subject: @project,
+                team: @project.team,
+                project: @project,
+                message_items: { project: @project.id,
+                                 visibility: message_visibility })
```
[the remark this answers is not in the document]
Comment (mlorb) [reply app/controllers/projects_controller.rb]: :+1:

File: app/controllers/protocols_controller.rb (hunk 17 above)
[the remark this answers is not in the document]
Comment (mlorb) [reply app/controllers/protocols_controller.rb]: :+1:

File: app/controllers/steps_controller.rb (hunk 1 above)
Comment (mlorb) [reply app/controllers/steps_controller.rb]: :+1:

File: app/controllers/steps_controller.rb (hunk 2 above)
Comment (mlorb) [reply app/controllers/steps_controller.rb]: :+1:

File: app/controllers/result_assets_controller.rb [hunk 20]
```diff
@@ -229,20 +209,33 @@ def create_multiple_results
...
-          type_of: :add_result,
-          user: current_user,
-          project: @my_module.experiment.project,
-          experiment: @my_module.experiment,
-          my_module: @my_module,
-          message: t('activities.add_asset_result',
-                     user: current_user.full_name,
-                     result: result.name)
-        )
+        Activities::CreateActivityService
```
[the remark this answers is not in the document]
Comment (mlorb) [reply app/controllers/result_assets_controller.rb]: :+1:

File: app/controllers/reports_controller.rb [hunk 21]
```diff
@@ -149,16 +131,13 @@ def destroy
...
-        type_of: :delete_report,
-        project: report.project,
-        user: current_user,
-        message: I18n.t(
-          'activities.delete_report',
-          user: current_user.full_name,
-          report: report.name
-        )
-      )
+      Activities::CreateActivityService
```
[the remark this answers is not in the document]
Comment (mlorb) [reply app/controllers/reports_controller.rb]: :+1:

File: spec/controllers/experiments_controller_spec.rb (hunk 4 above)
Comment (mlorb) [reply spec/controllers/experiments_controller_spec.rb]: :+1:

File: spec/controllers/project_comments_controller_spec.rb (hunk 5 above)
Comment (mlorb) [reply spec/controllers/project_comments_controller_spec.rb]: :+1:

File: spec/controllers/reports_controller_spec.rb (hunk 6 above)
Comment (mlorb) [reply spec/controllers/reports_controller_spec.rb]: :+1:

File: spec/controllers/result_assets_controller_spec.rb (hunk 7 above)
Comment (mlorb) [reply spec/controllers/result_assets_controller_spec.rb]: :+1:

File: spec/controllers/result_comments_controller_spec.rb (hunk 8 above)
Comment (mlorb) [reply spec/controllers/result_comments_controller_spec.rb]: :+1:

File: spec/controllers/result_tables_controller_spec.rb (hunk 10 above)
Comment (mlorb) [reply spec/controllers/result_tables_controller_spec.rb]: :+1:

File: spec/factories/result_comment.rb (hunk 12 above)
Comment (mlorb) [reply spec/factories/result_comment.rb]: :+1:

File: app/controllers/my_modules_controller.rb (hunk 14 above)
Comment (mlorb) [reply app/controllers/my_modules_controller.rb]: legacy code

File: app/controllers/my_modules_controller.rb (hunk 15 above)
Comment (mlorb) [reply app/controllers/my_modules_controller.rb]: legacy code

File: spec/controllers/result_comments_controller_spec.rb (hunk 9 above)
Comment (mlorb) [reply spec/controllers/result_comments_controller_spec.rb]: :+1:

File: spec/controllers/result_tables_controller_spec.rb (hunk 11 above)
Comment (mlorb) [reply spec/controllers/result_tables_controller_spec.rb]: :+1:

File: spec/factories/result_table.rb (hunk 13 above)
Comment (mlorb) [reply spec/factories/result_table.rb]: :+1:

**Pull request merged**